tests/postgres_tests/test_array.py

```diff
@@ -551,6 +552,71 @@ def test_grouping_by_annotations_with_array_field_param(self):
             1,
         )
 
+    def test_filter_by_array_subquery(self):
+        self.assertSequenceEqual(
+            NullableIntegerArrayModel.objects.alias(
+                same_sized_fields=ArraySubquery(
+                    NullableIntegerArrayModel.objects.filter(
+                        field__len=ArrayLenTransform(models.OuterRef('field'))
+                    ).values('field')
+                )
+            ).filter(same_sized_fields__len=2),
+            self.objs[0:2]
+        )
+
+    def test_annotated_array_subquery(self):
+        self.assertSequenceEqual(
+            NullableIntegerArrayModel.objects.annotate(
+                sibling_ids=ArraySubquery(
+                    NullableIntegerArrayModel.objects.exclude(
+                        id=models.OuterRef('id')
+                    ).values('id')
+                )
+            ).get(id=self.objs[0].id).sibling_ids,
+            [obj.id for obj in self.objs[1:]]
+        )
+
+    def test_group_by_with_annotated_array_subquery(self):
+        self.assertSequenceEqual(
+            NullableIntegerArrayModel.objects.annotate(
+                sibling_ids=ArraySubquery(
+                    NullableIntegerArrayModel.objects.exclude(
+                        id=models.OuterRef('id')
+                    ).values('id')
+                ),
+                sibling_count=models.Max(ArrayLenTransform('sibling_ids'))
+            ).values_list('sibling_count', flat=True),
+            [len(self.objs) - 1 for _ in self.objs]
+        )
+
+    def test_annotated_ordered_array_subquery(self):
+        self.assertSequenceEqual(
+            NullableIntegerArrayModel.objects.annotate(
+                ids=ArraySubquery(
+                    NullableIntegerArrayModel.objects.order_by('-id').values('id')
+                )
+            ).get(id=self.objs[0].id).ids,
```

Comment:
Why not `first()`? :thinking: 
```suggestion
            ).first().ids,
```

Reply:
A case of copy and paste! Thanks!